### PR TITLE
[DOCS-3350] Add control log volume and size guide

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do?

Adds a new guide for Observability Pipelines. Also adds link to the new guide from the Working with Data doc.

### Motivation

DOCS-3350

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/integrations/observability_pipelines/guide/control_log_volume_and_size.md [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+    tag: "Documentation"
+    text: "Working with data using Vector"
+  - link: /integrations/observability_pipelines/vector_configurations/
+    tag: Documentation
+    text: Learn more about Vector configurations
+---
+
+## Overview
+
+As your organization scales, your log volume grows, which means that the cost of ingesting and indexing your downstream services (for example, log management solutions, SIEMs, and so forth) also rises. This guide walks you through using Vector transforms to cut down on log volume and trim down the size of your logs to control your costs *before* data leaves your network.
```
Comment (barieom) [inline content/en/integrations/observability_pipelines/guide/control_log_volume_and_size.md:15]: fix to: "indexing in your downstream services"

Comment (barieom) [reply content/en/integrations/observability_pipelines/guide/control_log_volume_and_size.md:15]: fix to: "*before* data leaves your infrastructure or network."

**Review by barieom — APPROVED** [review]
Added a couple minor suggestions, thanks for driving this, @maycmlee !

File: content/en/integrations/observability_pipelines/guide/control_log_volume_and_size.md (hunk 1 above)
Comment (maycmlee) [inline content/en/integrations/observability_pipelines/guide/control_log_volume_and_size.md:15]:
```suggestion
As your organization scales, your log volume grows, which means that the cost of ingesting and indexing in your downstream services (for example, log management solutions, SIEMs, and so forth) also rises. This guide walks you through using Vector transforms to cut down on log volume and trim down the size of your logs to control your costs *before* data leaves your infrastructure or network. 
```

File: content/en/integrations/observability_pipelines/guide/control_log_volume_and_size.md [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+  }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+In the reduce transform, `group_by` is an ordered list of fields used to group events. In this example, the events are grouped by `host`, `pid`, and `tid` fields. 
+
+`merge_strategies` is a map of field names to custom merge strategies. There are [different merge strategies][10], ranging from `array`, where each value is appended to an array, to `sum`, which adds all numeric values. In this example, `concat_newline` is used, where each string value is concatenated then delimited by a newline.
```
Comment (cswatt) [inline content/en/integrations/observability_pipelines/guide/control_log_volume_and_size.md:361]:
```suggestion
`merge_strategies` is a map of field names to custom merge strategies. There are [different merge strategies][10], including `array`, where each value is appended to an array, and `sum`, which adds all numeric values. In this example, `concat_newline` is used, where each string value is concatenated, then delimited by a newline.
```

File: content/en/integrations/observability_pipelines/guide/control_log_volume_and_size.md [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+
+`starts_when` is a condition used to distinguish the first event of a transaction. If this condition resolves to `true` for an event, the previous transaction is flushed without this event and a new transaction is started. In this example, events with `.message` that do not match the `^[^\\s]` regular expression condition are reduced into a single event.
```
Comment (cswatt) [inline content/en/integrations/observability_pipelines/guide/control_log_volume_and_size.md:363]:
```suggestion
`starts_when` is a condition used to distinguish the first event of a transaction. If this condition resolves to `true` for an event, the previous transaction is flushed without this event, and a new transaction is started. In this example, events with `.message` that do not match the `^[^\\s]` regular expression condition are reduced into a single event. 
```

**Comment by maycmlee** [comment]
Thanks @cswatt for the review! I committed all the suggestions. Could you re-review?

**Review by cswatt — APPROVED** [review]

**Pull request merged**